Automattic/wp-calypso
# Stats: gate insights page and update traffic page gating

Related to p1733119153796319-slack-C068X5XJT3J

This PR aims to gate the Insights page and part of the Traffic page for WPCOM sites without the FEATURE_STATS_PAID feature.

## Proposed Changes

* Add `STATS_FEATURE_PAGE_INSIGHTS` and `STATS_FEATURE_PAGE_TRAFFIC` feature constants.
* Gate insights page under `stats/paid-wpcom-v3` feature flag
  * Hide nav item
  * Redirect to traffic page if try to reach it through URL
* Gate traffic page partially (replacing `STAT_TYPE_REFERRERS` check)

### Screenshot

![CleanShot 2024-12-02 at 20 30 38@2x](https://github.com/user-attachments/assets/038cb4c1-b138-4ddc-9dd4-9c1dfac7e369)

## Why are these changes being made?


p1HpG7-v6B-p2

## Testing Instructions



* Spin up a Calypso Live Branch
* Navigate to the Traffic page http://calypso.localhost:3000/stats/day/initsogarfree.wordpress.com using a free site
* Ensure the `Insights` tab is not visible
* Try navigating to the insights page directly: http://calypso.localhost:3000/stats/insights/initsogarfree.wordpress.com
* Ensure you are redirected back to the Traffic page

## Pre-merge Checklist
